libxmf/XmfMuxer.cpp

```diff
@@ -0,0 +1,497 @@
+#include "XmfMuxer.h"
+
+#include <iostream>
+#include <string>
+#include <mkvmuxer/mkvmuxer.h>
+#include <mkvmuxer/mkvmuxerutil.h>
+#include <mkvmuxer/mkvwriter.h>
+#include <mkvparser/mkvparser.h>
+#include <mkvparser/mkvreader.h>
+#include <hdr_util.hpp>
+#include <common/file_util.h>
+#include <errno.h>
+
+// Muxer structure holding necessary components
+struct xmf_webm_muxer {};
+
+bool CopyVideoProjection(const mkvparser::Projection& parser_projection,
+	mkvmuxer::Projection* muxer_projection);
+
+XmfWebMMuxer* XmfWebMMuxer_New() {
+	auto muxer = new XmfWebMMuxer();
```

Comment:
nit: I don't use auto as we're sticking to C as much as possible, use XmfWebMMuxer* as the explicit type

Reply:
updated